src/LinuxTracing/LibunwindstackUnwinderTest.cpp

```diff
@@ -0,0 +1,207 @@
+//     12a8:       48 01 44 24 08              add    %rax,0x8(%rsp)
+//     12ad:       83 44 24 04 01              addl   $0x1,0x4(%rsp)
+//     12b2:       83 7c 24 04 09              cmpl   $0x9,0x4(%rsp)
+//     12b7:       7e ea                       jle    12a3 <_Z10every_10usv+0x17>
+//     12b9:       48 8b 44 24 08              mov    0x8(%rsp),%rax
+//     12be:       48 83 c4 10 c3              add    $0x10,%rsp
+
+TEST(LibunwindstackUnwinder, DetectsFunctionWithFramePointerSet) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x124c, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_TRUE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsLeafFunction) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x122e, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_FALSE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsFunctionWithoutFramePointer) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_no_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x12ad, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_FALSE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsFramePointerNotSetAtPushRbp) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x1248, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_FALSE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsFramePointerNotSetAtMovRspToRbp) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x1249, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_FALSE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsFramePointerNotSetAtRet) {
+  auto unwinder = LibunwindstackUnwinder::Create();
+
+  auto maps = CreateFakeMapsEntry("target_fp");
+
+  std::optional<bool> has_frame_pointer_set_or_error =
+      unwinder->HasFramePointerSet(0x1279, maps->Get());
+
+  ASSERT_TRUE(has_frame_pointer_set_or_error.has_value());
+  EXPECT_FALSE(has_frame_pointer_set_or_error.value());
+}
+
+TEST(LibunwindstackUnwinder, DetectsFramePointerNotSetAtLeave) {
```

Comment:
I thought that at `leave` the frame pointer was still set because the instruction hasn't yet executed?